src/core/providers/ogr/qgsogrprovidermetadata.cpp

```diff
@@ -654,6 +655,117 @@ bool QgsOgrProviderMetadata::saveLayerMetadata( const QString &uri, const QgsLay
   throw QgsNotSupportedException( QObject::tr( "Storing metadata for the specified uri is not supported" ) );
 }
 
+QgsLayerMetadata QgsOgrProviderMetadata::loadLayerMetadata( const QString &layerUri, bool &found )
+{
+  QgsLayerMetadata metadata;
+  found = false;
+  const QVariantMap parts = decodeUri( layerUri );
+  const QString path = parts.value( QStringLiteral( "path" ) ).toString();
+  QString errorMessage;
+
+  if ( !path.isEmpty() && QFileInfo::exists( path ) )
+  {
+    const QFileInfo fi( path );
+    // if it is a gpkg, try reading metadata
+    if ( fi.suffix().compare( QLatin1String( "gpkg" ), Qt::CaseInsensitive ) == 0 )
+    {
+      const QString layerName = parts.value( QStringLiteral( "layerName" ) ).toString();
+      QgsOgrLayerUniquePtr userLayer;
+      userLayer = QgsOgrProviderUtils::getLayer( path, true, QStringList(), layerName, errorMessage, true );
+      if ( userLayer )
```

Comment:
Is there a test covering this already?

Reply:
Yes, I have added the test to the base test class in order to test all the implementations: https://github.com/qgis/QGIS/pull/55924/files#diff-3871ec4c64f2c18a6dbd113550e1c7250c488f3d939238053904714b544cfe77R144